jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/xloader/ProcBuildIndexX.java

```diff
@@ -189,6 +223,22 @@ private static long sort_build_index(Logger LOG, String datafile, DatasetGraph d
         InputStream input = IO.ensureBuffered(fromSortInputStream);
         // This thread - run builder.
         long count = indexBuilder(dsg, input, indexName);
+        try {
+            int exitCode = proc2.waitFor();
+            if ( exitCode != 0 ) {
+                String msg = IO.readWholeFileAsUTF8(proc2.getErrorStream());
+                String logMsg = String.format("Sort RC = %d : Error: %s", exitCode, msg);
+                Log.error(LOG, logMsg);
+                // ** Exit process
+                System.exit(exitCode);
+            }
+//            else
+//                LOG.info("Sort finished");
+            IO.close(toSortOutputStream);
+            IO.close(fromSortInputStream);
```

Comment:
Is it correct? Closing the streams only when succeed? Or should we close them in a finally?

Reply:
Can do and might as well. Anything other than a clean exit is "unstable" anyway.

If the sort failed, returns exit code != 0, then the JVM exits which closes the streams.